lambda/value_rec_compiler.ml

```diff
@@ -0,0 +1,560 @@
+(**************************************************************************)
+(*                                                                        *)
+(*                                 OCaml                                  *)
+(*                                                                        *)
+(*                       Vincent Laviron, OCamlPro                        *)
+(*                                                                        *)
+(*   Copyright 2023 OCamlPro SAS                                          *)
+(*                                                                        *)
+(*   All rights reserved.  This file is distributed under the terms of    *)
+(*   the GNU Lesser General Public License version 2.1, with the          *)
+(*   special exception on linking described in the file LICENSE.          *)
+(*                                                                        *)
+(**************************************************************************)
+
+open Lambda
+
+type size =
+  | Unreachable
+  | Dynamic
+  | Constant
+  | Function
+  | Regular_block of int
+  | Float_record of int
```

Comment:
The last two cases could be `Block of block_size`,  to avoid the redundancy with the type `static_block_size` below.